### PR TITLE
[INLONG-8445][Sort] Support running tests on both Flink 1.13 and Flink 1.15

### LINKED ISSUE #8445: [Improve][Sort] Support running tests on both Flink 1.13 and Flink 1.15
### Description

[Improve] Tests on sort module  (sort-common\sort-core\sort-end-to-end -tests) should support running on both flink1.13 and flink1.15

### InLong Component

InLong Sort

### Are you willing to submit PR?

- [X] Yes, I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://www.apache.org/foundation/policies/conduct)

### PR DESCRIPTION
**Pull request by haibo-duan**
### Prepare a Pull Request
*(Change the title refer to the following example)*

- [Sort] Support running tests on both Flink 1.13 and Flink 1.15
- Split the [end to end tests] module and add postgers case

*(The following *XYZ* should be replaced by the actual [GitHub Issue](https://github.com/apache/inlong/issues) number)*

- Fixes #8445 

### Motivation

*Explain here the context, and why you're making that change. What is the problem you're trying to solve?*

### Modifications

- Split the [end-to-end-tests] module into two versions, v1.13 and v1.15.
-  Add test cases that support reading postgers cdc data and writing to starrocks.

### Verifying this change

*(Please pick either of the following options)*

- [ ] This change is a trivial rework/code cleanup without any test coverage.

- [ ] This change is already covered by existing tests, such as:
  *(please describe tests)*

- [ ] This change added tests and can be verified as follows:

  *(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Documentation

  - Does this pull request introduce a new feature? (yes / no)
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a follow-up issue for adding the documentation

### CODE REVIEW

File: inlong-sort/sort-dist/pom.xml [hunk 1]
```diff
@@ -126,6 +126,10 @@
             <activation>
                 <activeByDefault>true</activeByDefault>
             </activation>
+            <properties>
+                <flink.version>1.13.5</flink.version>
+                <scala.binary.version>2.11</scala.binary.version>
+            </properties>
```
Comment (GanfengTan) [inline inlong-sort/sort-dist/pom.xml:132]: There are already global variables in the root pom.

File: inlong-sort/sort-dist/pom.xml [hunk 2]
```diff
@@ -145,19 +149,25 @@
...
                     <artifactId>flink-sql-avro</artifactId>
+                    <version>${flink.version}</version>
                 </dependency>
             </dependencies>
         </profile>
         <profile>
             <id>v1.15</id>
+            <properties>
+                <flink.version>1.15.4</flink.version>
+            </properties>
```
Comment (GanfengTan) [inline inlong-sort/sort-dist/pom.xml:170]: ditto

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.13/src/test/resources/log4j2-test.properties [hunk 3]
```diff
@@ -57,10 +57,10 @@ logger.tm.name=org.apache.flink.runtime.taskexecutor.TaskExecutor
 logger.tm.additivity=false
 
 logger.mysql=INFO, mysql
-logger.mysql.name=org.apache.inlong.sort.tests.utils.MySqlContainer
+logger.mysql.name=org.apache.inlong.sort.org.apache.inlong.sort.tests.utils.MySqlContainer
```
Comment (GanfengTan) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.13/src/test/resources/log4j2-test.properties:60]: The writing in this place is not very appropriate, please modify it.

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.13/src/test/resources/log4j2-test.properties [hunk 4]
```diff
@@ -57,10 +57,10 @@ logger.tm.name=org.apache.flink.runtime.taskexecutor.TaskExecutor
...
 logger.mysql.additivity=false
 
 logger.kafka=INFO, kafkaserver
-logger.kafka.name=org.apache.inlong.sort.tests.KafkaE2ECase
+logger.kafka.name=org.apache.inlong.sort.org.apache.inlong.sort.tests.KafkaE2ECase
 logger.kafka.additivity=false
```
Comment (GanfengTan) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.13/src/test/resources/log4j2-test.properties:65]: ditto

**Review by GanfengTan — APPROVED** [review]

**Comment by EMsnap** [comment]
Hi haibo-duan, thanks for your contribution, can you update the description for this pr and the modification you made ?

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/pom.xml [hunk 5]
```diff
@@ -0,0 +1,222 @@
...
+                            <goal>copy</goal>
+                        </goals>
+                        <phase>pre-integration-test</phase>
+                    </execution>
+                </executions>
+            </plugin>
+
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-surefire-plugin</artifactId>
```
Comment (EMsnap) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/pom.xml:191]: https://maven.apache.org/surefire/maven-failsafe-plugin/index.html is used for ITs, not surefire-plugin

**Comment by EMsnap** [comment]
It seems that the ITs you provide is not running in the pipeline, please check
image

File: inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/PostgresTest.java [hunk 6]
```diff
@@ -0,0 +1,219 @@
...
+import java.sql.Statement;
+import java.time.Duration;
+import java.util.Arrays;
+import java.util.List;
+import java.util.stream.Stream;
+
+/**
+ * End-to-end tests for sort-connector-postgres-cdc-v1.15 uber jar.
+ */
+public class PostgresTest extends FlinkContainerTestEnv {
```
Comment (EMsnap) [inline inlong-sort/sort-end-to-end-tests/sort-end-to-end-tests-v1.15/src/test/java/org/apache/inlong/sort/tests/PostgresTest.java:53]: Maybe use a different name to imply PostGre to Starrocks end to end test here

**Review by EMsnap — APPROVED** [review]

**Comment by EMsnap** [comment]
LGTM, Great Job !

**Pull request merged**